modules/stars.py

```diff
@@ -20,8 +20,263 @@
+
+    async def fetch(self):
+        query = """SELECT * FROM starboard_entries WHERE msg_id={}""".format(self.msg_id)
+
+        result = await self.db.fetchrow(query)
+
+        if result is None:
+            self.exists = False
+            return
+
+        self.exists = True
+        self.msg_id = result["msg_id"]
+        self.channel_id = result["channel"]
+        self.stars = result["stars"]
+        self.bot_message_id = result["bot_message_id"]
+        self.bot_content_id = result["bot_content_id"]
+
+
+class Starboard(core.Cog):
+    def __init__(self, bot: core.Bot):
+        self.bot = bot
+
+        self.remove_on_delete: bool = CONFIG.get("remove_on_delete")
+        self.entry_requirement: int = CONFIG.get("entry_requirement")
+        self.starboard_channel_id: int = CONFIG.get("starboard_channel_id")
+        self.pool: asyncpg.Pool[asyncpg.Record] = bot.pool
+
+    def get_star(self, stars: int):
+        if stars <= 2:
+            return "✨"
+        elif stars <= 4:
+            return "💫"
+        elif stars <= 6:
+            return "⭐"
+        else:
+            return "🌟"
+
+    async def add_entry(
+        self, message_id: int, bot_message_id: int, payload_channel_id: int, reactions: int, content_id: int
+    ):
+        query = """INSERT INTO starboard_entries VALUES (
+                    $1,
+                    $2,
+                    $3,
+                    $4,
+                    $5
+                )"""
+        await self.pool.execute(query, message_id, bot_message_id, payload_channel_id, reactions, content_id)
+
+    async def add_starer(self, user_id: int, message_id: int):
+        query = """
+                INSERT INTO starers VALUES (
+                    $1,
+                    $2
+                )"""
+
+        await self.pool.execute(query, user_id, message_id)
+
+    async def remove_starer(self, message_id: int, user_id: int):
+        query = """DELETE FROM starers WHERE msg_id = $1 AND user_id= $2"""
+        await self.pool.execute(query, message_id, user_id)
+
+    async def update_entry(self, reactions: int, message_id: int):
+        query = """UPDATE starboard_entries SET stars = $1 WHERE msg_id = $2"""
+        await self.pool.execute(query, reactions, message_id)
+
+    async def remove_entry(self, message_id: int):
+        query = """DELETE FROM starboard_entries WHERE msg_id= $1"""
+        await self.pool.execute(query, message_id)
+
+    async def clear_starers(self, message_id: int):
+        query = """DELETE FROM starers WHERE msg_id = $1"""
+        await self.pool.execute(query, message_id)
+
+    def get_formatted_time(self):
+        now = datetime.datetime.now()
+        time = now.strftime("%m/%d/%Y %I:%M %p")
+        return time
+
+    async def handle_star(self, payload: discord.RawReactionActionEvent):
+        time = self.get_formatted_time()
+        entry: StarboardEntry = StarboardEntry(self.pool, payload.message_id)
```

Comment:
Why don't we add a classmethod to `StarboardEntry`, like `from_record` or `from_query`?

```py
@classmethod
async def from_query(self, query: str) -> Self:
  ...
```

Then we don't need the analogous `.fetch()` call.